### ISSUE TITLE
Possible wrong on TestQuantile

### ISSUE THREAD

**Issue by iii-org-tw** [numpy/numpy#16856]
We are adding features on np.quantile(). 
When we ran runtest.py, we got two errors on the methods TestQuantile.test_quantile_monotonic and TestQuantile.test_quantile_monotonic_hypo in numpy/lib/tests/test_function_base.py.

    def test_quantile_monotonic(self):
        # GH 14685
        # test that the return value of quantile is monotonic if p0 is ordered
        p0 = np.arange(0, 1, 0.01)
        quantile = np.quantile(np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9,
                                         8, 8, 7]) * 0.1, p0)
        assert_equal(np.sort(quantile), quantile)

    @hypothesis.given(
            arr=arrays(dtype=np.float64,
                       shape=st.integers(min_value=3, max_value=1000),
                       elements=st.floats(allow_infinity=False, allow_nan=False,
                                          min_value=-1e300, max_value=1e300)))
    def test_quantile_monotonic_hypo(self, arr):
        p0 = np.arange(0, 1, 0.01)
        quantile = np.quantile(arr, p0)
        assert_equal(np.sort(quantile), quantile)

And I found that it would fail even keeping numpy source code unchanged.

### Reproducing code example:

```python
p0 = np.arange(0, 1, 0.01)
quantile = np.quantile(np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9,
                                 8, 8, 7]) * 0.1, p0)
np.testing.assert_equal(np.sort(quantile), quantile)
```
```python
arr=array([2.22044605e+284, 2.22044605e+284, 2.22044605e+284]),
p0 = np.arange(0, 1, 0.01)
quantile = np.quantile(arr, p0)
np.testing.assert_equal(np.sort(quantile), quantile)
```

### Error message:
```python
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-31-6281563fae61> in <module>
      2 quantile = np.quantile(np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9,
      3                                  8, 8, 7]) * 0.1, p0)
----> 4 np.testing.assert_equal(np.sort(quantile), quantile)

~/anaconda3/lib/python3.7/site-packages/numpy/testing/_private/utils.py in assert_equal(actual, desired, err_msg, verbose)
    347     from numpy.lib import iscomplexobj, real, imag
    348     if isinstance(actual, ndarray) or isinstance(desired, ndarray):
--> 349         return assert_array_equal(actual, desired, err_msg, verbose)
    350     msg = build_err_msg([actual, desired], err_msg, verbose=verbose)
    351 

~/anaconda3/lib/python3.7/site-packages/numpy/testing/_private/utils.py in assert_array_equal(x, y, err_msg, verbose)
    934     __tracebackhide__ = True  # Hide traceback for py.test
    935     assert_array_compare(operator.__eq__, x, y, err_msg=err_msg,
--> 936                          verbose=verbose, header='Arrays are not equal')
    937 
    938 

~/anaconda3/lib/python3.7/site-packages/numpy/testing/_private/utils.py in assert_array_compare(comparison, x, y, err_msg, verbose, header, precision, equal_nan, equal_inf)
    844                                 verbose=verbose, header=header,
    845                                 names=('x', 'y'), precision=precision)
--> 846             raise AssertionError(msg)
    847     except ValueError:
    848         import traceback

AssertionError: 
Arrays are not equal

Mismatched elements: 4 / 100 (4%)
Max absolute difference: 1.11022302e-16
Max relative difference: 1.23358114e-16
 x: array([0.   , 0.017, 0.034, 0.051, 0.068, 0.085, 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.108, 0.125, 0.142,...
 y: array([0.   , 0.017, 0.034, 0.051, 0.068, 0.085, 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.108, 0.125, 0.142,...
```
```python
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-31-6281563fae61> in <module>
      2 quantile = np.quantile(np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9,
      3                                  8, 8, 7]) * 0.1, p0)
----> 4 np.testing.assert_equal(np.sort(quantile), quantile)

~/anaconda3/lib/python3.7/site-packages/numpy/testing/_private/utils.py in assert_equal(actual, desired, err_msg, verbose)
    347     from numpy.lib import iscomplexobj, real, imag
    348     if isinstance(actual, ndarray) or isinstance(desired, ndarray):
--> 349         return assert_array_equal(actual, desired, err_msg, verbose)
    350     msg = build_err_msg([actual, desired], err_msg, verbose=verbose)
    351 

~/anaconda3/lib/python3.7/site-packages/numpy/testing/_private/utils.py in assert_array_equal(x, y, err_msg, verbose)
    934     __tracebackhide__ = True  # Hide traceback for py.test
    935     assert_array_compare(operator.__eq__, x, y, err_msg=err_msg,
--> 936                          verbose=verbose, header='Arrays are not equal')
    937 
    938 

~/anaconda3/lib/python3.7/site-packages/numpy/testing/_private/utils.py in assert_array_compare(comparison, x, y, err_msg, verbose, header, precision, equal_nan, equal_inf)
    844                                 verbose=verbose, header=header,
    845                                 names=('x', 'y'), precision=precision)
--> 846             raise AssertionError(msg)
    847     except ValueError:
    848         import traceback

AssertionError: 
Arrays are not equal

Mismatched elements: 4 / 100 (4%)
Max absolute difference: 1.11022302e-16
Max relative difference: 1.23358114e-16
 x: array([0.   , 0.017, 0.034, 0.051, 0.068, 0.085, 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.108, 0.125, 0.142,...
 y: array([0.   , 0.017, 0.034, 0.051, 0.068, 0.085, 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  ,
       0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.1  , 0.108, 0.125, 0.142,...
```

They may be caused by linear interpolation.
Should these functions test the samples by assert_allclose()?
### Numpy/Python version information:
numpy: '1.19.0'
python: 3.7

**Comment by eric-wieser** [comment]
Can you work out which four elements are the issue, and use `p0[[I, j, k, l]]` in place of `p0` in those examples?

**Comment by eric-wieser** [comment]
Are you sure you're running those tests again the latest numpy, and not an older numpy installed elsewhetr on your system?

**Comment by iii-org-tw** [comment]
> Can you work out which four elements are the issue, and use `p0[[I, j, k, l]]` in place of `p0` in those examples?
```
p0 = np.arange(0, 1, 0.01)
quantile = np.quantile(np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9,
                                 8, 8, 7]) * 0.1, p0)
index = np.sort(quantile).astype(str)!=quantile.astype(str)
p0[index]
array([0.89, 0.9 , 0.98, 0.99])
```
and
```
arr=array([2.22044605e+284, 2.22044605e+284, 2.22044605e+284]),
p0 = np.arange(0, 1, 0.01)
quantile = np.quantile(arr, p0)
index = np.sort(quantile).astype(str)!=quantile.astype(str)
p0[index]
array([0.  , 0.03, 0.04, 0.99])
```

**Comment by iii-org-tw** [comment]
> Are you sure you're running those tests again the latest numpy, and not an older numpy installed elsewhetr on your system?

I updated the version and now it's 1.19.0 but got the same errors.

**Comment by eric-wieser** [comment]
This was fixed in #16273, which is not part of the 1.19 release. If you run tests from master, you need to install from master too.
